### PR TITLE
add runtime arguments and hash partitioning to CountRandom; update docs

### PR DESCRIPTION
**Pull request by anew**
There is no Jira for this. But I added these to CountRandom for my manual testing of CDAP-2259 and thought it might be useful to add that to the example in the release, because we have no examples yet that use runtime arguments or hash partitioning.

### CODE REVIEW

**Comment by chtyim** [comment]
LGTM

File: cdap-docs/examples-manual/source/examples/count-random.rst [hunk 1]
```diff
@@ -40,11 +40,29 @@ The Flow contains three flowlets:
...
+
+- whether to emit events or not
+- to sleep for an additional delay after each event
 
 .. literalinclude:: /../../../cdap-examples/CountRandom/src/main/java/co/cask/cdap/examples/countrandom/RandomSource.java
    :language: java
-   :lines: 28-
+   :lines: 29-
+
+The *splitter* Flowlet emits four numbers four every number that it receives.
```
Comment (awholegunch) [inline cdap-docs/examples-manual/source/examples/count-random.rst]: four numbers for each number

File: cdap-docs/examples-manual/source/examples/count-random.rst [hunk 2]
```diff
@@ -40,11 +40,29 @@ The Flow contains three flowlets:
...
+
+.. literalinclude:: /../../../cdap-examples/CountRandom/src/main/java/co/cask/cdap/examples/countrandom/NumberSplitter.java
+   :language: java
+   :lines: 25-
+
+Note that this flowlet associates a hash value named *n* with every number that it emits. That allows the *counter*
+flowlet to use a hash partitioning strategy when consuming the numbers it receives. This ensures that there are no
```
Comment (awholegunch) [inline cdap-docs/examples-manual/source/examples/count-random.rst]: Flowlet

File: cdap-docs/examples-manual/source/examples/count-random.rst [hunk 3]
```diff
@@ -40,11 +40,29 @@ The Flow contains three flowlets:
...
+transaction conflicts if the flowlet is scaled to multiple instances:
```
Comment (awholegunch) [inline cdap-docs/examples-manual/source/examples/count-random.rst]: Flowlet

File: cdap-docs/examples-manual/source/examples/count-random.rst [hunk 4]
```diff
@@ -40,11 +40,29 @@ The Flow contains three flowlets:
    :language: java
    :lines: 25-
 
-The Flowlet (*source*) that generates random numbers every 1 millisecond:
+The *source* Flowlet generates random numbers every 1 millisecond. It can also be configured through runtime
+arguments:
+
+- whether to emit events or not
```
Comment (awholegunch) [inline cdap-docs/examples-manual/source/examples/count-random.rst]: whether to emit events or not; and

File: cdap-docs/examples-manual/source/examples/count-random.rst [hunk 5]
```diff
@@ -40,11 +40,29 @@ The Flow contains three flowlets:
...
+- to sleep for an additional delay after each event
```
Comment (awholegunch) [inline cdap-docs/examples-manual/source/examples/count-random.rst]: whether to sleep for an additional delay after each event.

**Comment by awholegunch** [comment]
A few comments. Fix and merge!

**Comment by anew** [comment]
thx for the comments. Fixed and will merge.

**Pull request merged**